### PR TITLE
doc: update release/README.md

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6920)

### CODE REVIEW

File: release/README.md [hunk 1]
```diff
@@ -86,25 +91,29 @@ discarded as they are uninteresting to our users.
...
 
-It is not recommended that you create the release branch before this PR is
-*merged*, but in some circumstances it might be needed, for example, if a large
-change that could destabilize the release is about to be merged, or if we want
-to create the release at an specific point in the revision history.
+In general, do not create the release branch before this PR is *merged*. We want
+to create the release from a stable point in the default branch (`main`), and
+we want this point to include the updated release notes and API baselines.
+There may be exceptions to this guideline, you are encouraged to use your own
+judgement.
```
Comment (dbolduc) [inline release/README.md:98]: judgment

File: release/README.md [hunk 2]
```diff
@@ -86,25 +91,29 @@ discarded as they are uninteresting to our users.
...
+[`release/release.sh`]: https://github.com/googleapis/google-cloud-cpp/blob/main/release/release.sh
 
 *No PR is needed for this step.*
 
 First run the following command -- which will *NOT* make any changes to any
 repos -- and verify that the output and *version numbers* look correct.
 
 ```bash
-$ release/release.sh googleapis/google-cloud-cpp
+$ `release/release.sh googleapis/google-cloud-cpp`
```
Comment (dbolduc) [inline release/README.md:116]: why?

File: release/README.md [hunk 3]
```diff
@@ -86,25 +91,29 @@ discarded as they are uninteresting to our users.
...
-to create the release at an specific point in the revision history.
+In general, do not create the release branch before this PR is *merged*. We want
+to create the release from a stable point in the default branch (`main`), and
+we want this point to include the updated release notes and API baselines.
+There may be exceptions to this guideline, you are encouraged to use your own
+judgement.
 
 ## Creating the release
 
+> :warning: the `release.sh` script was broken as of 2021-07.
```
Comment (dbolduc) [inline release/README.md:102]: should we open an issue?

Comment (coryan) [reply release/README.md:102]: There was one open, referenced here.

File: release/README.md (hunk 2 above)
Comment (coryan) [reply release/README.md]: I blame gremlins, fixed.

File: release/README.md (hunk 1 above)
Comment (coryan) [reply release/README.md]: Fixed, let's hope we have no customers used to British spelling.

File: release/README.md (hunk 3 above)
Comment (dbolduc) [reply release/README.md:102]: Thanks

**Review by dbolduc — APPROVED** [review]

**Pull request merged**